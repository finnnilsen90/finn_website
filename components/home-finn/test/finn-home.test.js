const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Home = require('../home-finn.js');

let HomeConst = new Home(); 

describe('component connection', () => {

    it('should be defined', () => {
        expect(Home).toBeDefined();
    })

})  
   
describe('<Home />', () => {

    const wrapper = render(<Home />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });


})

describe('Get hamburger menu', () => {

    let response = require('../../../data_files/base_menu.json')

    it('should return the json menu', () => {
        expect(response).toMatchSnapshot();
    })
})
