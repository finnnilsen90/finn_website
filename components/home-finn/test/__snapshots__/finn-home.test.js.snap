// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders components 1`] = `
<div
  class="containter"
>
  <div
    class="libcomp-hamburger_Container font"
  >
    <div
      class="ham_container"
    >
      <div
        class="hamburger"
      >
        <hr />
        <hr />
        <hr />
      </div>
    </div>
    <div
      class="dropdown_container"
    >
      <ul
        class="dropdown"
      >
        <li
          class="link ham_link"
        >
          <a
            class="a_link"
            href="/home"
          >
            Home Page
          </a>
        </li>
        <li
          class="link ham_link"
        >
          <a
            class="a_link"
            href="/resume"
          >
            Resume
          </a>
        </li>
        <li
          class="link ham_link"
        >
          <a
            class="a_link"
            href="/project"
          >
            Projects
          </a>
        </li>
        <li
          class="link ham_link"
        >
          <a
            class="a_link"
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="content-home_Container font"
  >
    <img
      class="profile"
      src="test-file-stub"
    />
    <p
      class="overview-finn"
    >
      My name is Finn Nilsen. I have been working in the tech industry for 5+ years in a variety of roles. Iâ€™ve been a software developer, digital marketer, and product manager. I have a passion for building great software products and leading technical teams. This website is an overview of my career so far and how I can help any company I work with grow.
    </p>
  </div>
  <div
    class="twocontent-home_Container font"
  >
    <h1
      class="resume_header"
    >
      <a
        href="/resume"
      >
        Resume
      </a>
    </h1>
    <img
      class="resume_icon"
      src="test-file-stub"
    />
    <div
      class="slider-home_Container font"
    >
      <div
        class="outer"
      >
        <div
          class="slide1 slide"
        >
          <a
            href="/resume"
          >
            <h2>
              Associate Director, Media and Technology
            </h2>
            <div
              class="slider_class"
            >
              Mogo Interactive
            </div>
          </a>
        </div>
        <div
          class="slide2 slide"
        >
          <a
            href="/resume"
          >
            <h2>
              Product Manager
            </h2>
            <div
              class="slider_class"
            >
              Mogo Interactive
            </div>
          </a>
        </div>
        <div
          class="slide3 slide"
        >
          <a
            href="/resume"
          >
            <h2>
              ChatScript Developer
            </h2>
            <div
              class="slider_class"
            >
              JustAnswer.com
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="button-one_Container"
    >
      <button
        class="button_style_one btn_mobile"
        href="/resume"
      >
        View Resume
      </button>
    </div>
  </div>
  <div
    class="threecontent-home_Container font font_color"
  >
    <h1
      class="project_header"
    >
      Projects
    </h1>
    <img
      class="project_icon"
      src="test-file-stub"
    />
    <ul
      class="project_list"
    >
      <li>
        <a
          class="project_link"
          href="/projects"
        >
          <h3>
            Hub by Mogo
          </h3>
          <p>
            Learn about how I managed the inital build and further development of Mogo's automation platform Hub.
          </p>
        </a>
      </li>
      <li>
        <a
          class="project_link"
          href="/projects"
        >
          <h3>
            Mogo's DMP
          </h3>
          <p>
            When Mogo onboarded a Data Managment Platform (DMP) I was part of the team that set up the connections and segment taxonomy.
          </p>
        </a>
      </li>
      <li>
        <a
          class="project_link"
          href="/projects"
        >
          <h3>
            JustAnswer's ChatBot
          </h3>
          <p>
            When JustAnswer was building out a ChatBot I was the main developer implementing the conversation logic.
          </p>
        </a>
      </li>
      <li>
        <a
          class="project_link"
          href="/projects"
        >
          <h3>
            Red Bull Youth America's Cup
          </h3>
          <p>
            I managed a team of sailors aiming to compete in the Red Bull Youth America's Cup.
          </p>
        </a>
      </li>
    </ul>
    <div
      class="button-one_Container"
    >
      <button
        class="button_style_one proj_btn_mobile"
        href="/project"
      >
        View Projects
      </button>
    </div>
  </div>
  <div
    class="footer-home_Container font"
  >
    <div
      class="logo_cont"
    >
      <a
        href="https://github.com/finnnilsen90"
      >
        <img
          class="github_logo"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/finn-nilsen-50815182/"
      >
        <img
          class="linkedin_logo"
          src="test-file-stub"
        />
      </a>
      <a
        href="mailto:fnilsen1051@gmail.com"
      >
        <img
          class="gmail_logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      class="link_cont"
    >
      <p
        class="git_link"
      >
        https://github.com/finnnilsen90
      </p>
      <p
        class="linkedin_link"
      >
        https://www.linkedin.com/in/finn-nilsen
      </p>
      <p
        class="gmail_link"
      >
        fnilsen1051@gmail.com
      </p>
    </div>
  </div>
</div>
`;

exports[`Get hamburger menu should return the json menu 1`] = `
Object {
  "admin": Object {
    "contact": Object {
      "className": "link",
      "link": "/contact",
      "text": "Contact",
    },
    "create_login": Object {
      "className": "create",
      "link": "/create_login",
      "text": "Create Login",
    },
    "form_submit": Object {
      "className": "form_submit",
      "link": "/form",
      "text": "Form",
    },
    "home_page": Object {
      "className": "link",
      "link": "/home_page",
      "text": "Home Page",
    },
    "logout": Object {
      "className": "link",
      "link": "/logout",
      "text": "Logout",
    },
    "projects": Object {
      "className": "link",
      "link": "/home_page",
      "text": "Projects",
    },
    "resume": Object {
      "className": "link",
      "link": "/resume",
      "text": "Resume",
    },
  },
  "logedin": Object {
    "contact": Object {
      "className": "link",
      "link": "/contact",
      "text": "Contact",
    },
    "home_page": Object {
      "className": "link",
      "link": "/home",
      "text": "Home Page",
    },
    "logout": Object {
      "className": "link",
      "link": "/logout",
      "text": "Logout",
    },
    "projects": Object {
      "className": "link",
      "link": "/projects",
      "text": "Projects",
    },
    "resume": Object {
      "className": "link",
      "link": "/resume",
      "text": "Resume",
    },
  },
  "not_login": Object {
    "home_page": Object {
      "className": "link",
      "link": "/home",
      "text": "Home Page",
    },
    "projects": Object {
      "className": "link",
      "link": "/project",
      "text": "Projects",
    },
    "resume": Object {
      "className": "link",
      "link": "/resume",
      "text": "Resume",
    },
  },
}
`;
